src/mpags-cipher.cpp

```diff
@@ -84,73 +61,72 @@ int main(int argc, char* argv[])
-    std::string inputText;
-
-    // Read in user input from stdin/file
-    // Warn that input file option not yet implemented
-    if (!inputFile.empty()) {
-        std::cerr << "[warning] input from file ('" << inputFile
-                  << "') not implemented yet, using stdin\n";
+    if (key.empty()){
+        std::cerr << "Please specify encryption or decryption key";
+        return 1;
     }
-
-    // loop over each character from user input
-    while (std::cin >> inputChar) {
-        // Uppercase alphabetic characters
-        if (std::isalpha(inputChar)) {
-            inputText += std::toupper(inputChar);
-            continue;
+    // initialise input and output strings
+    std::string inputText{""};
+    std::string outputText{""};
+    // Read in user input from stdin
+    if (inputFile.empty()) {
+        std::cerr << "No input file, specified, using std/in: \n";
+    
+        // Read out the transliterated text
+        char in_char{'x'};
+        
+        while (std::cin >> in_char) {
+            inputText += transformChar(in_char);
+        
         }
-
-        // Transliterate digits to English words
-        switch (inputChar) {
-            case '0':
-                inputText += "ZERO";
-                break;
-            case '1':
-                inputText += "ONE";
-                break;
-            case '2':
-                inputText += "TWO";
-                break;
-            case '3':
-                inputText += "THREE";
-                break;
-            case '4':
-                inputText += "FOUR";
-                break;
-            case '5':
-                inputText += "FIVE";
-                break;
-            case '6':
-                inputText += "SIX";
-                break;
-            case '7':
-                inputText += "SEVEN";
-                break;
-            case '8':
-                inputText += "EIGHT";
-                break;
-            case '9':
-                inputText += "NINE";
-                break;
+      
+    }
+    // Read in user input from file
+    else{
+        std::ifstream in_file {inputFile};
+        bool ok_to_read=in_file.good();
+        if (ok_to_read == true){
+            char inputChar {'x'};
+        
+            while (in_file >> inputChar){
+                inputText += transformChar(inputChar);
+            }
+            
+        }
+        else{
+            std::cerr << "[error] Could not read input file \n";
+            return 1;
         }
-
-        // If the character isn't alphabetic or numeric, DONT add it
     }
-
-    // Print out the transliterated text
-
-    // Warn that output file option not yet implemented
-    if (!outputFile.empty()) {
-        std::cerr << "[warning] output to file ('" << outputFile
-                  << "') not implemented yet, using stdout\n";
+    
+    
+   
+    // convert string key to some integer value
+    unsigned long shift_key;
+    shift_key = std::stoul(key, nullptr, 10);
```

Comment:
Can simplify this slightly:
```suggestion
    std::size_t shift_key { std::stoul(key) };
```
No need to specify the arguments of a function that have default values, unless you want to change them.